java/src/jmri/jmrix/can/TrafficController.java

```diff
@@ -32,4 +32,19 @@ public void setCanId(int canid) {
         _canid = canid;
     }
 
+    /**
+     * Set the CAN ID by numerical String value.
+     * @param canId String value of CAN ID.
+     */
+    public void setCanId( @javax.annotation.CheckForNull String canId) {
+        try {
+            setCanId(Integer.parseInt(canId));
+        } catch (NumberFormatException e) {
+            log.error("Cannot parse CAN ID \"{}\" - check your preference settings", canId, e);
+            log.error("Now using CAN ID {}",getCanid());
+        }
```

Comment:
If this happens, wouldn't it be good to alert the user with a message box showing the error, if JMRI is not headless?

I rarely looks at the logs unless I have a problem.

Reply:
Thanks for looking over @danielb987 ,
CBUS tries to keep CAN ID implementation details hidden from the user.
As the current implementation doesn't show a dialogue, no need to here either imho.